src/worker/network.rs

```diff
@@ -379,10 +394,19 @@ fn get_local_addr_port(
     let local_port = 49152 + (index % conns_per_addr as u64) as u16;
     debug!("addr {}, index {}", addr, index);
 
+    // conns_per_addr effectively groups connections together, one address per
+    // group with only port being different. addr_index represent current index
+    // inside the space of such groups.
     let mut addr_index = index / conns_per_addr as u64;
     let mut octets = addr.0;
 
-    info!("addr_index= {}", addr_index);
+    // The first resulting local address must not collide with the server
+    // addres, thus shift it by one right away.
+    if octets[3] != 255 {
+        octets[3] = octets[3].saturating_add(1);
```

Comment:
I don't think this is the best way of doing this. You could increment `addr` before calling this function or you could add `conns_per_addr` to `index` at the to or you could just add 1 to `octets[3]` and set carry to 1 if `octets[3]` is 255. I think `saturating_add` is not necessary if you have a check for `octets[3] != 255`. I don't think you should have that check. You should still increment the IP address in that case.

Reply:
I agree that having a saturating_add right after a check doesn't make much sense.

On top of that, we might need to improve error handling here, because what happens if the octet is not modified? Will we get 2 colliding IP addresses? Does that work or will it cause something to break down the line?

w.r.t carrying the overflow over, that could cause the IP address to move to a different network, so that might also not be the desired behavior (although no handling of mask value is done here, so 🤷🏻 )